src/app/sub.rs

```diff
@@ -69,6 +71,29 @@ where
             no_sort,
         } = self;
 
+        // If one of the files is on HDD, set thread number to 1
+        let disks = Disks::new_with_refreshed_list();
+
+        if files
+            .iter()
+            .filter_map(|file| fs::canonicalize(file).ok())
+            .any(|path| {
+                let mount_points: Vec<_> = disks.iter().map(|disk| disk.mount_point()).collect();
+                if let Some(mount_point) = find_mountpoint(&path, &mount_points) {
+                    disks.iter().any(|disk| {
+                        matches!(disk.kind(), DiskKind::HDD) && disk.mount_point() == mount_point
+                    })
+                } else {
+                    false
+                }
+            })
```

Comment:
I think this whole expression inside `if` should be a private function that takes `get_disk_kind: impl Fn(&Disk) -> DiskKind` as an argument. And there should be a test for it too.

Reply:
@KSXGitHub I have moved this whole expression to a separate private function `detect_hdd_in_files()`, however it is difficult to write test for it.  Both `sysinfo::Disk` & `sysinfo::Disks` has private internal fields, which means we can't create `Disk` or `Disks` structs by ourselves.

Reply:
@Integral-Tech The solution is to make `Disk` a generic type parameter: https://github.com/KSXGitHub/parallel-disk-usage/pull/269/commits/30883d7213cd9172c240da7a033673ee39964539

Reply:
@KSXGitHub Can I add tests for `path_is_in_hdd()` and `any_path_is_in_hdd()` by myself now?

Reply:
Sure. Go ahead.

Reply:
@KSXGitHub Test for `path_is_in_hdd()` has been added. As for `any_path_is_in_hdd()`, it uses `std::fs::canonicalize()` which will return an error if the corresponding path doesn't exist.

Reply:
@KSXGitHub Done :)